language-api/src/main/java/de/jplag/util/FileUtils.java

```diff
@@ -0,0 +1,141 @@
+        removeBom(reader, charset);
+        return reader;
+    }
+
+    /**
+     * Reads the contents of a file into a single string.
+     * @param file The file to read
+     * @return The files content as a string
+     * @throws IOException If an IO error occurs
+     * @see FileUtils#openFileReader(File)
+     */
+    public static String readFileContent(File file) throws IOException {
+        try (BufferedReader reader = openFileReader(file)) {
+            return reader.lines().collect(Collectors.joining(System.lineSeparator()));
+        }
+    }
+
+    /**
+     * Removes the byte order mark from the beginning of the stream, if it exists and the charset is a UTF* charset. For
+     * details see: <a href="https://en.wikipedia.org/wiki/Byte_order_mark">Wikipedia</a>
+     * @param reader The reader to remove the bom from
+     * @throws IOException If an IO error occurs.
+     */
+    private static void removeBom(BufferedReader reader, Charset charset) throws IOException {
+        if (charset.name().toUpperCase().startsWith("UTF")) {
+            reader.mark(10);
+            if (reader.read() != '\uFEFF') {
+                reader.reset();
+            }
+        }
+    }
+
+    /**
+     * Detects the charset of a file. Prefer using {@link #openFileReader(File)} or {@link #readFileContent(File)} if you
+     * are only interested in the content.
+     * @param file The file to detect
+     * @return The most probable charset
+     * @throws IOException If an IO error occurs
+     */
+    public static Charset detectCharset(File file) throws IOException {
+        try (InputStream stream = new BufferedInputStream(new FileInputStream((file)))) {
+            return detectCharset(stream);
+        }
+    }
+
+    /**
+     * Detects the most probable charset over the whole set of files.
+     * @param files The files to check
+     * @return The most probable charset
+     */
+    public static Charset detectCharsetFromMultiple(Collection<File> files) {
+        Map<String, List<Integer>> charsetValues = new HashMap<>();
+
+        files.stream().map(it -> {
+            try (InputStream stream = new BufferedInputStream(new FileInputStream(it))) {
+                return detectAllCharsets(stream);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }).forEach(matches -> {
+            Set<String> remaining = new HashSet<>(Set.of(CharsetDetector.getAllDetectableCharsets()));
+            for (CharsetMatch match : matches) {
+                charsetValues.putIfAbsent(match.getName(), new ArrayList<>());
+                charsetValues.get(match.getName()).add(match.getConfidence());
+                remaining.remove(match.getName());
+            }
+            remaining.forEach(it -> {
+                charsetValues.putIfAbsent(it, new ArrayList<>());
+                charsetValues.get(it).add(0);
+            });
+        });
```

Comment:
I'd suggest not using lambdas here (esp. lambda blocks). Simply use normal for loops. that could increase readability :)